packages/core/admin/ee/server/services/review-workflows/entity-service-decorator.js

```diff
@@ -43,9 +44,36 @@ const decorator = (service) => ({
     const data = { ...opts.data };
     if (isNull(data[ENTITY_STAGE_ATTRIBUTE])) {
       delete data[ENTITY_STAGE_ATTRIBUTE];
+      return service.update.call(this, uid, entityId, { ...opts, data });
     }
 
-    return service.update.call(this, uid, entityId, { ...opts, data });
+    const entity = await service.findOne.call(this, uid, entityId, {
+      populate: {
+        [ENTITY_STAGE_ATTRIBUTE]: {
+          populate: {
+            workflow: true,
+          },
+        },
+      },
+    });
+    const previousStageId = entity?.[ENTITY_STAGE_ATTRIBUTE]?.id ?? null;
```

Comment:
Minor thing, and I don't know if we have any consensus here, but I would prefer to be more explicit on the code and have things like this encapsulated in functions
```suggestion
const previousStage  = getEntityStage(id)
```

And this is personal, It makes it easy to read and follow the business logic, and you can look at the query details later if you need.